pandas/core/index.py

```diff
@@ -114,17 +114,21 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None, fastpath=False,
         if fastpath:
             return cls._simple_new(data, name)
 
+        if is_categorical_dtype(data) or is_categorical_dtype(dtype):
+            return CategoricalIndex(data, copy=copy, name=name, **kwargs)
+
+        from pandas.tseries.index import DatetimeIndex
```

Comment:
I would leave all of the imports where they were, no reason to actually import them until they are used (see below)